xray/grpc_client.go

```diff
@@ -0,0 +1,17 @@
+package xray
+
+import (
+	"context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func UnaryClientInterceptor(name string) grpc.UnaryClientInterceptor {
```

Comment:
How do we plan to propagate context ? We might need to pass ctx with the name argument.

Reply:
Hi,

Yeah, I agree that would be better to propagate existing context instead of creating a dummy segment, I have the following implementation for my project:

`func addTraceData(ctx context.Context, req interface{}) (context.Context, error) {
        // should contains traceID and segment in context
	seg := xray.GetSegment(ctx)
	if seg == nil || seg.TraceID == "" {
		return ctx, fmt.Errorf("No xray segment found")
	}
	subCtx, subSeg := xray.BeginSubsegment(ctx, "grpc-client")
	defer subSeg.Close(nil)

	md, ok := metadata.FromOutgoingContext(subCtx)
	if !ok {
		md = metadata.MD{}
	}
        // we can iterate over all values from the context and pass it too
	md.Set("traceID", seg.TraceID)
	md.Set("parentID", seg.ID)

	outCtx := metadata.NewOutgoingContext(subCtx, md)
	xray.AddMetadata(subCtx, "Request", req)

	return outCtx, nil
}`

and then use it in Interceptor before invoker.

But in order to trace all the info, would be better either to implement something similar as https://github.com/grpc-ecosystem/grpc-opentracing or enable support for OpenTracing in Xray.